File: csharp/PoS_before/PoS_before/POSTerminal/PosTerminal.cs
using System;

namespace POSTerminal
{
    public class PosTerminal
    {
        private CCVerifier ccVerifier;

        public PosTerminal()
        {
            this.ccVerifier = CCVerifier.getInstance();
        }
        
        void VerifySale(double amount)
        {
            if (!ccVerifier.approveCharge(amount))
            {
                throw new InvalidChargeException();
            }
        }
    }

    internal class InvalidChargeException : Exception
    {
        public InvalidChargeException() : 
            base("Charge exceeds single transaction amount or credit limit")
        {
        }
    }
}

File: java/PoS_after/src/main/java/POSTerminal.java
public class POSTerminal {
    ICCVerifier ccVerifier;

    POSTerminal() {
        this.ccVerifier = CCVerifier.getInstance();
    }

    POSTerminal(ICCVerifier ccVerifier) {
        this.ccVerifier = ccVerifier;
    }

    void verifySale(double amount) throws InvalidCharge {
        if (!ccVerifier.approveCharge(amount)) {
            throw new InvalidCharge();
        }
    }

    public static class InvalidCharge extends Throwable {
        public InvalidCharge() {
            super("Charge exceeds single transaction amount or credit limit");
        }
    }
}
